evidence/reactor.go

```diff
@@ -82,10 +82,15 @@ func (evR *Reactor) Receive(chID byte, src p2p.Peer, msgBytes []byte) {
 	case *ListMessage:
 		for _, ev := range msg.Evidence {
 			err := evR.evpool.AddEvidence(ev)
-			if err != nil {
-				evR.Logger.Info("Evidence is not valid", "evidence", msg.Evidence, "err", err)
+			switch err.(type) {
+			case ErrInvalidEvidence:
+				evR.Logger.Error("Evidence is not valid", "evidence", msg.Evidence, "err", err)
 				// punish peer
 				evR.Switch.StopPeerForError(src, err)
+			case nil:
```

Comment:
```go
case ErrEvidenceAlreadyStored:
	evR.Logger.Debug("Evidence already exist", "evidence", msg.Evidence)
```

Reply:
This doesn't work, I don't think, because ErrEvidenceAlreadyStored isn't a type